OMCompiler/SimulationRuntime/cpp/Core/Math/ArrayOperations.cpp

```diff
@@ -492,6 +492,17 @@ T sum_array (const BaseArray<T>& x)
   return val;
 }
 
+template <typename T>
+T product_array(const BaseArray<T>& x)
+{
+  const T* data = x.getData();
+  size_t nelems = x.getNumElems();
+  T val = 1;
+  for (size_t i = 0; i < nelems; i++)
+    val *= *data++;
```

Comment:
This should be more in line with `sum_array`.
```suggestion
  T val = std::accumulate(data, data + x.getNumElems(), T(1), std::multiplies<T>());
```